### DISCUSSION TITLE
Alpha-testing `py4ami`

### DISCUSSION THREAD

**Discussion by ShweataNHegde** [petermr/petermr#54] [General]
I'm testing it on Windows 10

## Cloning Repo
```
git clone https://github.com/petermr/pyami.git
```
While cloning, I ran into this error: 
```
Cloning into 'pyami'...
remote: Enumerating objects: 6860, done.
remote: Counting objects: 100% (1352/1352), done.
remote: Compressing objects: 100% (536/536), done.
remote: Total 6860 (delta 877), reused 1267 (delta 794), pack-reused 5508
Receiving objects: 100% (6860/6860), 191.14 MiB | 654.00 KiB/s, done.
Resolving deltas: 100% (3181/3181), done.
error: invalid path 'temp/ipcc_html/chapter_4:.html'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```
I happen to have encountered this problem before. Windows doesn't like `:` in file names. I fixed the error by running:
```
git restore --source=HEAD :/
```
## Running `ami_pdf`
```
python -m py4ami.ami_pdf --help
```
Ran into this: 
```
running PDFArgs main
module_stem ami_pdf
 sys.argv ['D:\\pyami\\py4ami\\ami_pdf.py', '--help']
sys.argv ['D:\\pyami\\py4ami\\ami_pdf.py', '--help']
Traceback (most recent call last):
  File "D:\pyami\py4ami\ami_pdf.py", line 1741, in main
    pdf_args.parse_and_process()
  File "D:\pyami\py4ami\util.py", line 362, in parse_and_process
    self.create_arg_parser()
  File "D:\pyami\py4ami\ami_pdf.py", line 742, in create_arg_parser
    self.parser.add_argument("--flow", type=bool, nargs=1, help="create flowing HTML (heuristics)", default=True)
  File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1386, in add_argument
    return self._add_action(action)
  File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1749, in _add_action
    self._optionals._add_action(action)
  File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1590, in _add_action
    action = super(_ArgumentGroup, self)._add_action(action)
  File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1400, in _add_action
    self._check_conflict(action)
  File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1539, in _check_conflict
    conflict_handler(action, confl_optionals)
  File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1548, in _handle_conflict_error
    raise ArgumentError(action, message % conflict_string)
argparse.ArgumentError: argument --flow: conflicting option string: --flow

***Cannot run pyami***; see output for errors: argument --flow: conflicting option string: --flow
```
I can't immediately see why I get this error. Any pointers?

**Comment by petermr** [comment]
On Sun, Sep 4, 2022 at 6:06 AM Shweata N. Hegde ***@***.***>
wrote:

> I'm testing it on Windows 10
> Cloning Repo
>
> git clone https://github.com/petermr/pyami.git
>
> While cloning, I ran into this error:
>
> Cloning into 'pyami'...
> remote: Enumerating objects: 6860, done.
> remote: Counting objects: 100% (1352/1352), done.
> remote: Compressing objects: 100% (536/536), done.
> remote: Total 6860 (delta 877), reused 1267 (delta 794), pack-reused 5508
> Receiving objects: 100% (6860/6860), 191.14 MiB | 654.00 KiB/s, done.
> Resolving deltas: 100% (3181/3181), done.
> error: invalid path 'temp/ipcc_html/chapter_4:.html'
> fatal: unable to checkout working tree
> warning: Clone succeeded, but checkout failed.
> You can inspect what was checked out with 'git status'
> and retry with 'git restore --source=HEAD :/'
>
> I happen to have encountered this problem before. Windows doesn't like :
> in file names. I fixed the error by running:
>
> git restore --source=HEAD :/
>
> Thanks. I have to remove this from the autogeneration.

> Running ami_pdf
>
> python -m py4ami.ami_pdf --help
>
> Ran into this:
>
> running PDFArgs main
> module_stem ami_pdf
>  sys.argv ['D:\\pyami\\py4ami\\ami_pdf.py', '--help']
> sys.argv ['D:\\pyami\\py4ami\\ami_pdf.py', '--help']
> Traceback (most recent call last):
>   File "D:\pyami\py4ami\ami_pdf.py", line 1741, in main
>     pdf_args.parse_and_process()
>   File "D:\pyami\py4ami\util.py", line 362, in parse_and_process
>     self.create_arg_parser()
>   File "D:\pyami\py4ami\ami_pdf.py", line 742, in create_arg_parser
>     self.parser.add_argument("--flow", type=bool, nargs=1, help="create flowing HTML (heuristics)", default=True)
>   File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1386, in add_argument
>     return self._add_action(action)
>   File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1749, in _add_action
>     self._optionals._add_action(action)
>   File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1590, in _add_action
>     action = super(_ArgumentGroup, self)._add_action(action)
>   File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1400, in _add_action
>     self._check_conflict(action)
>   File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1539, in _check_conflict
>     conflict_handler(action, confl_optionals)
>   File "C:\Users\shweata\AppData\Local\Programs\Python\Python38\lib\argparse.py", line 1548, in _handle_conflict_error
>     raise ArgumentError(action, message % conflict_string)
> argparse.ArgumentError: argument --flow: conflicting option string: --flow
>
> ***Cannot run pyami***; see output for errors: argument --flow: conflicting option string: --flow
>
> I can't immediately see why I get this error. Any pointers?
>

I think "--flow" was included twice. I changed it but you may have to use
`pmr9` branch  . I'll try to fix this later today.

> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/54>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS5FSSJTZDWAYL63RMDV4QU3VANCNFSM6AAAAAAQEFLPVI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK
